resources/session02/http_server.py

```diff
@@ -1,12 +1,73 @@
 import socket
 import sys
+import mimetypes
+import os.path
+
+
+def response_ok(body=b"this is a pretty minimal response", mimetype=b"text/plain"):
+    """returns a basic HTTP response"""
+    mimetype = b"content-type:" + mimetype
+    resp = []
+    resp.append(b"HTTP/1.1 200 OK")
+    resp.append(mimetype)
+    resp.append(b"")
+    resp.append(body)
+    return b"\r\n".join(resp)
+
+
+def response_method_not_allowed():
+    """returns a 405 Method Not Allowed response"""
+    resp = []
+    resp.append("HTTP/1.1 405 Method Not Allowed")
+    resp.append("")
+    return "\r\n".join(resp).encode('utf8')
+
+
+def response_not_found():
+    """returns a 404 Not Found response"""
+    return b"HTTP/1.1 404 Not Found"
+
+
+def parse_request(request):
+    first_line = request.split("\r\n", 1)[0]
+    try:
+        method, uri, protocol = first_line.split()
+    # Chrome seems to be sending a zero-length request that breaks the split.
+    except ValueError:
+        print('The client tried to break me.  Request was: \'{}\''.format(request))
+        method = ''
+    if method != "GET":
+        raise NotImplementedError("We only accept GET")
+    return uri
+
+
+def resolve_uri(uri):
+    """Return appropriate content and a mime type"""
+    # Return the contents of a file if it exists
+    uri = os.getcwd() + uri
```

Comment:
This means that everything about the server, including the code that makes the server operate, is open to being read by a web request.  That's not such a good idea from a security standpoint.  The 'home directory' should really be the `webroot` folder in the homework resources.  Can you figure out how to make that happen?